### DISCUSSION TITLE
COCO - Ultralytics YOLOv8 Docs

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#2675] [Docs]
# COCO - Ultralytics YOLOv8 Docs

Learn about the COCO dataset, designed to encourage research on object detection, segmentation, and captioning with standardized evaluation metrics.

https://docs.ultralytics.com/datasets/detect/coco/

**Comment by navidTerraNova** [comment]
According to training2017.txt in the dataset file, 56600 of the total 118k images are being used for training and when you want to change that and use more data it seems like according to the label folder only same 56600  number of 118k images are labeled. is that right? or am I missing something...

**Reply by pderrenger** [reply]
@navidTerraNova hello! It sounds like there might be a misunderstanding regarding the dataset and its annotations. The COCO dataset typically includes labels for all the images provided in the training set. If you're seeing a discrepancy in the number of labeled images, it could be due to several reasons:

1. **Partial Download:** Perhaps not all the annotation files were downloaded successfully.
2. **Dataset Configuration:** The dataset configuration file (`coco.yaml`) might be pointing to a subset of the full annotations.
3. **Custom Split:** Sometimes, for quicker experiments, a smaller subset of the full dataset is used, which might be the case with the `train2017.txt` you're referring to.

To resolve this, you can:

- Ensure that the full dataset has been downloaded, including all annotation files.
- Check the `coco.yaml` file and verify the paths to the annotation files.
- If you're using a custom dataset split, adjust the `train2017.txt` to include all desired images, and ensure corresponding labels are available.

If you continue to face issues, it might be helpful to review the dataset structure in the directory and confirm that all expected annotation files are present. The full COCO dataset should indeed have labels for all 118k images in the training set. If you need further assistance, the documentation at https://docs.ultralytics.com can provide more detailed guidance on dataset preparation and usage. 😊📚

**Comment by kvnptl** [comment]
Hello, this is not instance segmentation; it's semantic segmentation. For instance, when I examined the person class (0), all instances were the same color, lacking unique IDs for the same class. Could you please make the necessary updates?

It should be like this. [img source](https://blog.roboflow.com/difference-semantic-segmentation-instance-segmentation/)
![image](https://github.com/ultralytics/ultralytics/assets/47410011/935c3e05-52ef-421f-968d-a9bf481cabde)

**Reply by pderrenger** [reply]
@kvnptl hello! Thank you for bringing this to our attention. It seems there might be a misunderstanding regarding the segmentation type in the dataset or model you are using. In YOLOv8, we support only instance segmentation. Output images are colored by class, but output results objects definitely return per-instance segments.

If you need further assistance with YOLOv8 and instance segmentation, please refer to our documentation for detailed guidance on how to train, validate, and predict with instance segmentation models. 😊📚
